bolt12/offers-test.json

```diff
@@ -0,0 +1,494 @@
+    "valid": false,
+    "bolt12": "lno1pgpgqsgkyypqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqs"
+  },
+  {
+    "description": "Malformed: truncated offer_paths",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3qqgpzcssyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsz"
+  },
+  {
+    "description": "Malformed: zero num_hops in blinded_path",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3qqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsqzcssyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsz"
+  },
+  {
+    "description": "Malformed: truncated onionmsg_hop in blinded_path",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3qqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqspqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqgkyypqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqs"
+  },
+  {
+    "description": "Malformed: bad first_node_id in blinded_path",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3qqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqspqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqgqzcssyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsz"
+  },
+  {
+    "description": "Malformed: bad blinding in blinded_path",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3qqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcpqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqgqzcssyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsz"
+  },
+  {
+    "description": "Malformed: bad blinded_node_id in onionmsg_hop",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3qqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqspqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqgqzcssyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsz"
+  },
+  {
+    "description": "Malformed: truncated issuer UTF-8",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3yqvqzcssyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqsz"
+  },
+  {
+    "description": "Malformed: invalid issuer UTF-8",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3yq5qgytzzqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqg"
+  },
+  {
+    "description": "Malformed: invalid offer_node_id",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3vggzqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvpsxqcrqvps"
+  },
+  {
+    "description": "Contains type >= 80",
+    "valid": false,
+    "bolt12": "lno1pgz5znzfgdz3vggzqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgp9qgr0u2xq4dh3kdevrf4zg6hx8a60jv0gxe0ptgyfc6xkryqqqqqqqq"
+  },
+  {
+    "description": "Contains unknown feature 22",
+    "valid": false,
+    "bolt12": "lno1pgx9getnwss8vetrw3hhyucvqdqqqqqkyypwa3eyt44h6txtxquqh7lz5djge4afgfjn7k4rgrkuag0jsd5xvxg"
+  },
```

Comment:
LDK happily parses this, which I think is correct as per:

>   - if `offer_features` contains unknown _even_ bits that are non-zero:
>     - MUST NOT respond to the offer.
>     - SHOULD indicate the unknown bit to the user.

Consider a web site that parses BOLT 12 offers. We'd wouldn't want parsing to fail if there is an even, unknown feature bit. We still want to display the information within the offer to the user, which is different from responding to the offer.

See also earlier conversation: https://github.com/lightning/bolts/pull/798#issuecomment-1678055298

Reply:
In any case, 22 is not unknown, it is `option_anchors`.